api/lib/logger.ts

```diff
@@ -177,9 +183,128 @@ export function module(module: string): ModuleLogger {
  * Setup all loggers starting from config
  */
 export function setupAll(config: DeepPartial<GatewayConfig>) {
+	if (cleanJob) {
+		clearInterval(cleanJob)
+		cleanJob = undefined
+	}
+
 	logContainer.loggers.forEach((logger: ModuleLogger) => {
 		logger.setup(config)
 	})
 }
 
+let cleanJob: NodeJS.Timeout
+
+function setupCleanJob(settings: DailyRotateFileTransportOptions) {
+	if (cleanJob) {
+		return
+	}
+
+	let maxFilesMs: number
+	let maxFiles: number
+	let maxSizeBytes: number
+
+	const logger = module('LOGGER')
+
+	// convert maxFiles to milliseconds
+	if (settings.maxFiles !== undefined) {
+		const matches = settings.maxFiles.toString().match(/(\d+)([dhm])/)
+
+		if (settings.maxFiles) {
+			const value = parseInt(matches[1])
+			const unit = matches[2]
+			switch (unit) {
+				case 'd':
+					maxFilesMs = value * 24 * 60 * 60 * 1000
+					break
+				case 'h':
+					maxFilesMs = value * 60 * 60 * 1000
+					break
+				case 'm':
+					maxFilesMs = value * 60 * 1000
+					break
+			}
+		} else {
+			maxFiles = Number(settings.maxFiles)
+		}
+	}
+
+	if (settings.maxSize !== undefined) {
+		// convert maxSize to bytes
+		const matches2 = settings.maxSize.toString().match(/(\d+)([kmg])/)
+		if (matches2) {
+			const value = parseInt(matches2[1])
+			const unit = matches2[2]
+			switch (unit) {
+				case 'k':
+					maxSizeBytes = value * 1024
+					break
+				case 'm':
+					maxSizeBytes = value * 1024 * 1024
+					break
+				case 'g':
+					maxSizeBytes = value * 1024 * 1024 * 1024
+					break
+			}
+		} else {
+			maxSizeBytes = Number(settings.maxSize)
+		}
+	}
+
+	// clean up old log files based on maxFiles and maxSize
+
+	const filePathRegExp = new RegExp(
+		path.basename(settings.filename).replace(/%DATE%/g, '([^.]+)'),
+	)
+
+	const logsDir = path.dirname(settings.filename)
+
+	const clean = async () => {
+		try {
+			logger.info('Cleaning up log files...')
+			const files = await readdir(logsDir)
+			const logFiles = files.filter(
+				(file) =>
+					file !== settings.symlinkName && filePathRegExp.test(file),
+			)
+
+			const logFilesStats = await Promise.all(
+				logFiles.map(async (file) => ({
+					file,
+					stats: await stat(path.join(logsDir, file)),
+				})),
+			)
+
+			logFilesStats.sort((a, b) => a.stats.mtimeMs - b.stats.mtimeMs)
+
+			// sort by mtime
+
+			let totalSize = 0
+			let deletedFiles = 0
+			for (const { file, stats } of logFilesStats) {
+				const filePath = path.join(logsDir, file)
+				totalSize += stats.size
+
+				const fileMs = stats.mtimeMs
+
+				const shouldDelete =
+					(maxSizeBytes && totalSize > maxSizeBytes) ||
+					(maxFiles && logFiles.length - deletedFiles > maxFiles) ||
+					(maxFilesMs && fileMs && Date.now() - fileMs > maxFilesMs)
+
+				if (shouldDelete) {
+					logger.info(`Deleting log file: ${filePath}`)
+					await unlink(filePath)
```

Comment:
you might wamt to ave an own try/catch here, else it stops on first file it can not delete ... and in fact "File not exsts error "is a success :)

Reply:
Yeah, agree 👍